gsa/scripts/gsa_prepare_data.sh

```diff
@@ -1,40 +1,43 @@
 #!/bin/bash    
 
+[ -z ${ATLAS_EXPS+x} ] && echo "Env var ATLAS_EXPS for the path to atlas experiments needs to be defined." && exit 1
+[ -z ${ATLAS_GSA_FTP_DIR+x} ] && echo "Env var ATLAS_GSA_FTP_DIR for the path to the atlas ftp gsa folder needs to be defined." && exit 1
+
+WORKDIR="${ATLAS_GSA_WORKDIR:-${TMPDIR}/gsa}"
+
 # This script creates database files (one per organism in Atlas) to be used in the on-the-fly gene set enrichment against differentially expressed genes in each comparisons in a given Atlas organism
 
 scriptDir=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
-source ${scriptDir}/../../../bash_util/generic_routines.sh
 
 # quit if not prod user
-check_prod_user
+atlas-bash-util check_prod_user
 if [ $? -ne 0 ]; then
     exit 1
 fi
 
-pushd $ATLAS_EXPS
-wDir=~/tmp/gsa
-mkdir -p $wDir
-rm -rf $wDir/*
-gsaFTPDir=/ebi/ftp/pub/databases/arrayexpress/data/atlas/gsa/
+pushd $ATLAS_EXPS > /dev/null
+mkdir -p $WORKDIR
+rm -rf $WORKDIR/*
 
 echo "About to assemble organism-experiments tuples"
-grep organism contrastdetails.tsv | grep -P '\torganism\t' | awk -F"\t" '{print $6"\t"$1}' | sort | uniq | sed 's| |_|g' > ${wDir}/exps.aux
-awk '$1 = tolower($1)' ${wDir}/exps.aux > ${wDir}/exps.aux.lc && mv ${wDir}/exps.aux.lc ${wDir}/exps.aux
-perl -pi -e 's|^oryza_sativa |oryza_sativa_japonica_group |g' ${wDir}/exps.aux
-perl -pi -e 's|^oryza_sativa_japonica |oryza_sativa_japonica_group |g' ${wDir}/exps.aux
+grep organism contrastdetails.tsv | grep -P '\torganism\t' | awk -F"\t" '{print $6"\t"$1}' | sort | uniq | sed 's| |_|g' > ${WORKDIR}/exps.aux
+awk '$1 = tolower($1)' ${WORKDIR}/exps.aux > ${WORKDIR}/exps.aux.lc && mv ${WORKDIR}/exps.aux.lc ${WORKDIR}/exps.aux
+perl -pi -e 's|^oryza_sativa |oryza_sativa_japonica_group |g' ${WORKDIR}/exps.aux
+perl -pi -e 's|^oryza_sativa_japonica |oryza_sativa_japonica_group |g' ${WORKDIR}/exps.aux
 
 echo "About to assemble tsv input lists for gsa_prepare_data.R" 
-for l in $(cat ${wDir}/exps.aux); do
+
+cat ${WORKDIR}/exps.aux | while read -r l; do
     o=`echo $l | awk '{print $1}'`
     e=`echo $l | awk '{print $2}'`
     ls $ATLAS_EXPS/$e/${e}*-analytics.tsv > /dev/null
     if [ $? -eq 0 ]; then 
        ls $ATLAS_EXPS/$e/$e-configuration.xml > /dev/null
        if [ $? -eq 0 ]; then 
```

Comment:
maybe shorter?
```suggestion
       if [ -f $ATLAS_EXPS/$e/$e-configuration.xml ]; then 
```

Reply:
Thanks! I wasn't being very critical of the code.